roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -82,7 +102,13 @@ def is_active():
 
     @abstractmethod
     def run(self):
-        """Executes a check, normally implemented as a module."""
+        """Executes a check against a host and returns a result hash similar to Ansible modules.
+
+        Actually the directon ahead is to record state in the attributes and
+        not bother building a result hash. Instead, return an empty hash and let
+        the action plugin fill it in. Or raise an OpenShiftCheckException.
+        Returning a hash may become deprecated if it does not prove necessary.
```

Comment:
I would go for getting rid of the old way if possible to avoid confusion

Reply:
Well it's a lot of code and tests to refactor. But maybe I can get it done quickly enough.